src/test/java/org/folio/rest/utils/UserSummaryUtils.java

```diff
@@ -0,0 +1,39 @@
+package org.folio.rest.utils;
+
+import java.util.stream.IntStream;
+
+import org.folio.domain.OpenLoan;
+import org.folio.domain.UserSummary;
+import org.folio.repository.UserSummaryRepository;
+import org.folio.rest.TestBase;
+
+import io.vertx.ext.unit.TestContext;
+
+public class UserSummaryUtils extends TestBase {
+
+  protected void checkUserSummary(String summaryId, UserSummary userSummaryToCompare,
+    TestContext context, UserSummaryRepository userSummaryRepository) {
+
+    UserSummary userSummary = waitFor(userSummaryRepository.get(summaryId)).orElseThrow(() ->
+      new AssertionError("User summary was not found: " + summaryId));
+
+    context.assertEquals(userSummaryToCompare.getUserId(), userSummary.getUserId());
+    context.assertEquals(0, userSummaryToCompare.getOutstandingFeeFineBalance().compareTo(
+      userSummary.getOutstandingFeeFineBalance()));
+    context.assertEquals(userSummaryToCompare.getNumberOfLostItems(),
+      userSummary.getNumberOfLostItems());
+    context.assertEquals(userSummaryToCompare.getOpenLoans().size(),
+      userSummary.getOpenLoans().size());
+
+    IntStream.range(0, userSummary.getOpenLoans().size())
+      .forEach(i -> {
+        OpenLoan openLoan = userSummary.getOpenLoans().get(i);
+        OpenLoan openLoanToCompare = userSummaryToCompare.getOpenLoans().get(i);
```

Comment:
Is it possible that the two summaries will contain the same set of loans but in different order?

Reply:
Theoretically it's possible, so this is a very strict validation. But it was intended for tests and it works for all of the current test cases. If we ever need to allow open loans in different order, this method can be improved to support it.